### ISSUE TITLE
Report incompatible `TypedDict` field override

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6588]
When a `TypedDict` class derives from (extends) another `TypedDict` class, a field that is redeclared in the extended class should be flagged as an error if the `Required` or `NotRequired` attribute doesn't match.

```python
from typing import NotRequired, Required, TypedDict

class A(TypedDict):
    x: Required[int]

class B(A):
    x: NotRequired[int] # Should be an error
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.339, which I just published. It will also be included in a future release of pylance.
